### ISSUE TITLE
swcMinifier change the logical behavior of a some code

### ISSUE THREAD

**Issue by paoloricciuti** [vercel/next.js#41527]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: Ubuntu 20.04.0 LTS Tue Oct 18 2022 18:22:36 GMT+0200 (Central European Summer Time)
    Binaries:
      Node: 16.14.2
      npm: 7.17.0
      Yarn: 1.22.19
      pnpm: 7.13.0
    Relevant packages:
      next: 12.3.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

The code at line 92 of the toolbar of react-quill

https://github.com/quilljs/quill/blob/20f4c45258af235f5be4a2b7f5411c9b5a9e620e/modules/toolbar.ts#L92

it's meant to format the text based on the selected element inside a `<select>` in the toolbar. This get's correctily minified by react-quill build process:

```javascript
s=!u.hasAttribute("selected")&&(u.value||!1)
```

but this code will later be re-minified by next.js swcMinifier in a logically similar but behavior-ally different code

```javascript
s=!u.hasAttribute("selected")&&!!u.value
```

This lead to the unusability of react-quill inside Next.JS unless you disable the swcMinifier (given that the former minification always cast the value to a boolean the changes will never apply).

This is also a silent bug since there are no error and in development everything works fine. 

### Expected Behavior

the minified code should have the same logic as the unminified one

### Link to reproduction

https://stackblitz.com/edit/react-quill-repro?file=next.config.js

### To Reproduce

1. Visit [this stackblitz](https://stackblitz.com/edit/react-quill-repro?file=next.config.js)
2. In the integrated terminal run `npm run dev`
3. Try to write something and change either the font, the size or the color. Everything will work fine.
4. Stop the dev server with CTRL+C
5. run `npm run build`
6. run `npm run start`
7. Try to write something and change either the font, the size or the color. This times neither will work.
8. Change next.config.js removing `swcMinify: true`
9. run `npm run build`
10. run `npm run start`
11. Try to write something and change either the font, the size or the color. This times everything will work fine.

**Comment by steveluscher** [comment]
I was literally just drumming up a POC that demonstrates this issue.

1. Clone this repo: https://github.com/steveluscher/swc-minification-error-poc
2. `yarn && yarn build`

Observe that this code:

```typescript
/** Implement {@link Layout#encode|encode} for {@link Structure}.
 *
 * If `src` is missing a property for a member with a defined {@link
 * Layout#property|property} the corresponding region of the buffer is
 * left unmodified. */
encode(src, b, offset = 0) {
    const firstOffset = offset;
    let lastOffset = 0;
    let lastWrote = 0;
    for (const fd of this.fields) {
        let span = fd.span;
        lastWrote = (0 < span) ? span : 0;
        if (undefined !== fd.property) {
            const fv = src[fd.property];
            if (undefined !== fv) {
                lastWrote = fd.encode(fv, b, offset);
                if (0 > span) {
                    /* Read the as-encoded span, which is not necessarily the
                        * same as what we wrote. */
                    span = fd.getSpan(b, offset);
                }
            }
        }
        lastOffset = offset;
        offset += span;
    }
    /* Use (lastOffset + lastWrote) instead of offset because the last
        * item may have had a dynamic length and we don't want to include
        * the padding between it and the end of the space reserved for
        * it. */
    return (lastOffset + lastWrote) - firstOffset;
}
```

Gets minified to this:

```typescript
encode(t,e,r=0){let n=0,i=0;for(let o of this.fields){let s=o.span;if(i=0<s?s:0,void 0!==o.property){let u=t[o.property];void 0!==u&&(i=o.encode(u,e,r),0>s&&(s=o.getSpan(e,r)))}n=r,r+=s}return n+i-r}
```

Notice how `firstOffset` is erased in the minified code, and replaced with `offset`, breaking the code.

**Comment by ijjk** [comment]
Hi, this should be fixed in the latest canary of Next.js `v12.3.2-canary.32`, please update and give it a try!

**Comment by steveluscher** [comment]
Awesome! So fixed in 12.3.2. Can you tell me what version the bug snuck in at?
